packages/commons/src/interop-token/interopTokenService.ts

```diff
@@ -22,15 +36,27 @@ export class InteropTokenGenerator {
   private kmsClient: KMSClient;
 
   constructor(
-    private config: TokenGenerationConfig &
-      Partial<SessionTokenGenerationConfig>
+    private config: Partial<AuthorizationServerTokenGenerationConfig> &
+      Partial<TokenGenerationConfig> &
+      Partial<SessionTokenGenerationConfig>,
```

Comment:
I'm afraid the extended use of `Partial` may lead to issues by weakening the type checking.
Plus, there is no guarantee that there is no overlapping between fields in types (that may contain different values).

In case, we may proceed with this change in the PR, and then open a new PR right away for the refactor

Reply:
I agree that this is clearly not optimal. I opted for this solution to have custom env vars for the authorization server and avoid touching the config of the other packages that use this service. Let's do a refactor in a following PR.
By the way, right now there is no overlapping in field names. For example: `GENERATED_INTEROP_TOKEN_KID`, `INTERNAL_JWT_KID`, `GENERATED_JWT_KID`. One solution to simplify the logic might be to make them have the exact same name, but let's discuss it further.